### ISSUE TITLE
reorder_imports does not respect comments

### ISSUE THREAD

**Issue by malbarbo** [rust-lang-nursery/rustfmt#1373]
```rust
// Comment 1
use std::mem;
use std::marker;

// Comment 2
use std::iter;

pub fn main() {}
```

rustfmt produces

```rust
// Comment 1


// Comment 2
use std::iter;
use std::marker;
use std::mem;

pub fn main() {}
```

The comment 1 lost their context. I would expect something like:

```rust
// Comment 1
use std::marker;
use std::mem;

// Comment 2
use std::iter;

pub fn main() {}
```

Maybe related with https://github.com/rust-lang-nursery/rustfmt/issues/1137

**Comment by ishitatsuyuki** [comment]
This needs some kind of discussion.

Currently there's a test that expects the behavior that comments moves together with the code below; this one doesn't follow the rule.

Example:
```
/// sticky
use b;
use a;
```
to:
```
use a;
/// sticky
use b;
```

**Comment by nrc** [comment]
@ishitatsuyuki I'm not clear - are you saying the example is expected but doesn't happen or happens but shouldn't? It looks correct to me (that is the 'sticky' comment sticks to the import of `b`).

**Comment by ishitatsuyuki** [comment]
@nrc this is the current behavior and it clearly conflicts with the example posted by OP.

**Comment by nrc** [comment]
Ah ok, so I guess the question is whether the first comment should be special-cased and kept at the top of the imports instead of sticking to the import it is 'on top of'? At the moment, it seems that the comment is left and newlines are inserted? There also seems to be some confusion about whether we maintain grouping of imports.

So, I think that
* we should fix the extra newlines - that looks like just a bug
* If there is a newline between comments and the first `use`, leave the comment where it is. If not, the comment should stick to the `use`
* No grouping for of imports for now (at least, not by using newlines).

**Comment by topecongiro** [comment]
Triage: the first issue is fixed, closing in favor of #2185.
